rasa/core/actions/action.py

```diff
@@ -336,6 +339,7 @@ def event_for_successful_execution(
             confidence=prediction.max_confidence,
             action_text=self.action_text,
             hide_rule_turn=prediction.hide_rule_turn,
+            metadata=prediction.action_metadata,
```

Comment:
How did you know that this `ActionEndtoEndResponse` method needs to be updated? By searching for all usages when an `ActionExecuted` gets returned?

Reply:
Yes, basically. We need to make sure that `ActionExecuted`'s metadata is updated with `PolicyPrediction`'s metadata — which means we need to check where the constructor for `ActionExecuted` is being called and update it accordingly.